### DISCUSSION TITLE
Resolving services

### DISCUSSION THREAD

**Discussion by itsmenotjeff** [dotnet/maui#5774] [Q&A] [answered]
What is the syntax for resolving a service registered with the Maui Builder? Can't seem to find it anywhere.

If I have a DialogService:
``````
builder.Services.AddSingleton<IDialogService, DialogService>();
``````

How do I resolve this without using dependency injection? Trying to work around [this issue](https://github.com/dotnet/maui/discussions/5263) by resolving a dependency manually in a view model that would normally use DI

**Comment by pictos** [comment, marked as answer]
@itsmenotjeff I believe this will solve your issue
https://github.com/davidortinau/WeatherTwentyOne/blob/main/src/WeatherTwentyOne/Services/ServiceExtensions.cs

**Reply by itsmenotjeff** [reply]
THANK YOU SO MUCH. I was looking all over google for this and for the life of me couldn't find it. Should have figured to check out the demo app.
